cfn.template

```diff
@@ -189,16 +201,96 @@ Resources:
+                      <table>
+                        <tr>
+                          <th>Region</th>
+                  EOF
+
+                  for template in $templates; do
+                  cat <<EOF >>index.html
+                          <th>$(basename $template)</th>
+                  EOF
+                  done
+
+                  cat <<EOF >>index.html
+                        </tr>
+                  EOF
+
+                  for region in $regions; do
+                    export AWS_DEFAULT_REGION="${region}"
+                    export BUCKET="${bucket_name_prefix}-${region}"
+
+                    cat <<EOF >>index.html
+                        <tr>
+                          <th>${region}</th>
+                  EOF
 
-                    # Sync to latest
-                    if [ "$version" != "latest" ]; then
-                      echo "Syncing $version to latest"
-                      aws s3 rm --recursive "s3://${BUCKET}/${project}/{latest}"
-                      aws s3 cp --acl "$acl" "s3://${BUCKET}/${project}/${version}/main.template" "s3://${BUCKET}/${project}/latest/main.template"
-                    fi
+                    for template in $templates; do
+                      name="${template%.*}"
+                      url="https://s3.amazonaws.com/${BUCKET}/${project}/${version}/$(basename $template)"
 
-                    echo "https://${BUCKET}.s3-website-${region}.amazonaws.com/${project}/$version/index.html"
+                      cat <<EOF >>index.html
+                          <td>
+                            <a href="https://console.aws.amazon.com/cloudformation/home?region=${region}#/stacks/new?stackName=${name}&templateURL=${url}">
+                              <img alt="Launch Stack" src="https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png" />
+                            </a>
+                          </td>
+                  EOF
+                    done
+
+                    cat <<EOF >>index.html
+                        </tr>
+                  EOF
                   done
+
+                  # Close the index file
+                  cat <<EOF >>index.html
+                      </table>
+                      <p class="footer">Generated by cfn-publish on $(date -u)</p>
+                    </body>
+                  </html>
+                  EOF
+
+                  # Upload the website
+                  export AWS_DEFAULT_REGION="${first_region}"
+                  export BUCKET="${bucket_name_prefix}-${first_region}"
+                  aws s3api put-bucket-website --bucket "$BUCKET" --website-configuration '{"IndexDocument": {"Suffix": "index.html"}}'
+                  aws s3 cp --acl "$acl" index.html "s3://${BUCKET}/${project}/${version}.html"
+                  aws s3 cp --acl "$acl" index.html "s3://${BUCKET}/${project}/index.html"
+
+                  echo "http://${BUCKET}.s3-website-${first_region}.amazonaws.com/${project}/index.html"
```

Comment:
This could be the wrong URL for some regions (see https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints) but as this whole HTML file is an undocumented feature of cfn-publish for now anwyay... meh? :)

Note for future: We should consider reworking cfn-publish into step functions triggered by codebuild.